CMakeLists.txt

```diff
@@ -113,6 +113,25 @@ if(CODECOVERAGE)
     message( "-- Building with llvm Code Coverage Tools")
 endif()
 
+if((NOT USE_CRYPTOPP_HASH) AND (NOT USE_OPENSSL_SHA_256) AND (NOT USE_OPENSSL_SHA3_256))
+    message(FATAL_ERROR "None of the cryptographic hashing libraries are enabled.")
+elseif(USE_OPENSSL_SHA_256 AND USE_OPENSSL_SHA3_256)
```

Comment:
Why is this check needed? As per the code, it seems that we are using a prioritized approach.
#if defined USE_CRYPTOPP_HASH
using Digest = DigestHolder<CryptoppDigestCreator>;
#elif defined USE_OPENSSL_SHA_256
using Digest = DigestHolder<OpenSSLDigestCreator<SHA2_256> >;
#elif defined USE_OPENSSL_SHA3_256
using Digest = DigestHolder<OpenSSLDigestCreator<SHA3_256> >;
#endif
So if everything is enabled then we set Cryptopp. So I think setting all the macros ON is not an issue as such. But keeping this logic will add more maintenance overhead in the future. So we will just set one single in our build env, and we are tacking this as per priority. Only the first check is good enough as we need at least one to be kept.

Reply:
Addressed https://github.com/arc-vmware/concord-bft/pull/2/commits/be7b4d3f850359b235887ae82cbe8f4c29a080c4